### PR TITLE
Comment section

### LINKED ISSUE #8: [4pt]  - Create feature: add new comment - Student B
Implement the "add new comment" form on the Comments pop up according to the wireframe:

Screenshot 2021-06-27 at 22 32 36

When the user clicks on the "Comment" button, the data is recorded in the [Involvement API](https://www.notion.so/microverse/Involvement-API-869e60b5ad104603aa6db59e08150270) and the screen is updated.

### LINKED ISSUE #11: [3pt]  - Display comments for a given item on the Comments pop-up - Student B
When the popup loads, the webapp retrieves data from the [Involvement API](https://www.notion.so/microverse/Involvement-API-869e60b5ad104603aa6db59e08150270) to show the item comments.

Screenshot 2021-06-27 at 22 21 00

This task **does not** include the counter of the comments.

### PR DESCRIPTION
**Pull request by johnie261**
I did the following:

- Created a comment section and styled it.
- Created a form to input a name, comment, and submit button.
- Implemented the add new comment form on the Comments pop-up according to the wireframe:

### CODE REVIEW

File: src/popup/modal.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            <div class="box"> Type : ${character.type} </div>
+            <div class="box"> Gender : ${character.gender} </div>
+        </div>
+        <div class="comment-box">
+          <h2 class="title"> Comments (2) </h2>
+          <div class="comment-container">
+            <p> 15/03/2023 John: I'd love to buy it so much </p>
+            <p> 15/03/2023 John: I'd love to buy it </p>
+          </div>
+        </div>
```
Comment (NicolasAndrehh) [inline src/popup/modal.js]: I kindly recommend you to check the requirements again, you are showing the comments section, that great but the comments of that section should be from the data from the API of this endpoint /apps/:app_id/comments, you can add some comments using the following structure in the body of the POST request: 

{
    "item_id": "item1",
    "username": "Jane",
    "comment": "Hello"
}

You can use the involvement API class and add a method to get comments, remember that the post comment method is not a requirement for this activity you'll do it on an upcoming activity

You can take a look at the get likes method to create the get comments method they suppose to be alike

**Review by NicolasAndrehh — CHANGES_REQUESTED** [review]
## Status: Changes Required ♻️
Congratulations @johnie261, you did a great job but there are some things that can be improved!

## Highlights
- Very clean code
- Good GitFlow
- Good job with the media queries

## Required Changes 
There are a few changes that I request before approving the PR:

- Check the comments below

*Feel free to leave a comment tagging me if any changes is not clear at all*

**Comment by johnie261** [comment]
@NicolasAndrehh thanks for the suggestion man

File: src/popup/comment.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+export default class InvolvementApi {
+  constructor() {
+    this.url = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/hWosAqiVeXtujgpmRoG5/comments?item_id=item1';
+  }
```
Comment (NicolasAndrehh) [inline src/popup/comment.js]: At the end of the URL where it says 'item1' should be replaced with the item id, that id should be passed as a parameter in the addComments(itemId) method, and then when you use the fetch method it should be something like this `fetch('${this.url}${itemId}', { the rest of the code })` that way if you pass the id 6 as a parameter you'll get all the comments from the item 6, and for checking if it working you'll need to post some comments for the specifics ids that you'll check and remember to use postman to check if your POSTS were successful

**Comment by johnie261** [comment]
Something happened while resolving conflict,,,let me solve it

File: src/popup/comment.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        }),
+        headers: { 'Content-type': 'application/json; charset=UTF-8' },
+      });
+      if (!response.ok) {
+        throw new Error(`Error: ${response.statusText}`);
+      }
+    } catch (error) {
+      throw new Error(`Error:${error}`);
+    }
+  }
```
Comment (NicolasAndrehh) [inline src/popup/comment.js]: Excellent implementation of the methods, but I recommend you put those methods on the InvolvementAPI that was already created on `modules/involvementAPI.js`.
After doing that may you would need to change some imports in your other files

**Review by NicolasAndrehh — CHANGES_REQUESTED** [review]
## Status: Changes Required ♻️
Congratulations @johnie261, you did a great job but there are some things that can be improved!

## Highlights
- Good job with the involvementAPI methods
- Great GitFlow

## Required Changes 
There are a few changes that I request before approving the PR:
- I like that you're experimenting with box-shadow, but I kindly recommend you follow the design of the page, try to put the comment button alike the homepage buttons, stick to the colors and padding, you can use the border shadow as a hover effect, and experiment with the width or position of the button
- Check the comments below for more changes

*Feel free to leave a comment tagging me if any changes are not clear at all*

**Review by NicolasAndrehh — APPROVED** [review]
## PR Accepted ✅
Congratulations @johnie261, you did a great job, there's nothing else to say more than let's merge it! 🎉

### Highlights
- Awesome job implementing the comments for each Character
- Very clean code

**Pull request merged**